### PR TITLE
Create/drop tables and schemas

### PR DESCRIPTION
**Pull request by henrik1**
Added functionality for creation of schemas and tables without having to store data while doing so. This is particularly useful when running migrations. Also added functionality for dropping schemas and tables with optional cascade.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -226,6 +293,78 @@ Massive.prototype.documentTableSql = function(tableName){
   return sql;
 };
 
+
+Massive.prototype.dropDocumentTable = function(table, options, next) {
```
Comment (dmfay) [inline index.js]: I think calling it `dropTable` instead loses the consistency with `createDocumentTable` but makes it more obvious that it's generally applicable -- there's nothing specific to document tables in the functionality.

File: index.js [hunk 2]
```diff
@@ -217,6 +217,73 @@ var MapToNamespace = function(entity, collection) {
   db[collection].push(entity);
 };
 
+var RemoveFromNamespace = function(db, table, collection) {
```
Comment (dmfay) [inline index.js]: Some dead code in here, and it looks like you're never calling this function with the `collection` parameter set -- were you going for dynamically dropping and removing views?

File: index.js [hunk 3]
```diff
@@ -217,6 +217,73 @@ var MapToNamespace = function(entity, collection) {
...
+  if(splits.length > 1) {
+    schemaName = splits[0];
+    tableName = splits[1];
+  } else {
+    schemaName = "public";
+    tableName = table;
+  }
+
+  if(schemaName === "public") {
+    db[table] = undefined;
```
Comment (dmfay) [inline index.js]: You want to `delete` stuff here: setting `undefined` doesn't actually remove the key so `Object.keys()`, `hasOwnProperty`, etc still see something there.

**Comment by dmfay** [comment]
Good stuff here, thanks! I left a few comments on some implementation details :)

File: index.js (hunk 1 above)
Comment (henrik1) [inline index.js]: Happy to change the name. Though I do disagree that there is no specific functionality as these tables are created specifically as document store tables. Still in this context I completely agree that changing the name to createTable makes sense. 👍

Comment (dmfay) [inline index.js]: Oh, no, `createDocumentTable` should definitely stay as-is -- it's just `dropDocumentTable` will work equally well on a document or any other table, so there's no need to call it out specifically.

File: index.js (hunk 2 above)
Comment (henrik1) [inline index.js]: Yep, but I can remove it for now since it is not sufficiently tested and also not in use yet. 👍

File: index.js (hunk 3 above)
Comment (henrik1) [inline index.js]: 👍

File: index.js (hunk 1 above)
Comment (henrik1) [inline index.js]: It's late night, my bad :)

**Comment by dmfay** [comment]
Great! Can you update the name of drop_document_table.sql so it's consistent with the new function name, and then I think it's good to go :)

**Comment by aabenoja** [comment]
Can we get a release of this, please? It's pretty disorienting that the documentation for the new functions are there but aren't available on the latest release.

**Comment by robconery** [comment]
Yep, done - 2.3 is up

**Pull request merged**